### PR TITLE
Release procedure

### PR DESCRIPTION
**Pull request by cczhu**
Added release procedure to the documentation.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,12 @@
+2.0 (unreleased)
+================
+
+Other Changes and Additions
+---------------------------
+
+- Added release procedure to the documentation.  [#268]
```
Comment (mhvk) [inline CHANGES.rst]: The sentence above stating that this is about adding a new format, now no longer makes sense!

Comment (cczhu) [reply CHANGES.rst]: I don't have a strong opinion about this, but Astropy has its release procedure under ["Developer Documentation"](http://docs.astropy.org/en/stable/#developer-documentation), which makes sense to me (though they also have more stuff in general, so maybe it looks less jarring).

**Comment by cczhu** [comment]
I don't have a strong opinion about this, but Astropy has its release procedure under ["Developer Documentation"](http://docs.astropy.org/en/stable/#developer-documentation), which makes sense to me (though they also have more stuff in general, so maybe it looks less jarring).

File: CHANGES.rst (hunk 1 above)
Comment (cczhu) [reply CHANGES.rst]: Now addressed.

**Pull request merged**